File: trees/1.go
package main

import "fmt"

// Node comment
type Tree1 struct {
	left  *Tree1
	val   int
	right *Tree1
}

func insert1(node *Tree1, data int) *Tree1 {
	if node == nil {
		return &Tree1{nil, data, nil}
	}

	if data == node.val {
		return node
	}

	if data < node.val {
		node.left = insert1(node.left, data)
	} else {
		node.right = insert1(node.right, data)
	}
	return node
}

func inOrderTraverse1(node *Tree1) {
	if node == nil {
		return
	}
	inOrderTraverse1(node.left)
	fmt.Print(node.val, " ")
	inOrderTraverse1(node.right)
}

func preOrderTraverse1(node *Tree1) {
	if node == nil {
		return
	}
	fmt.Print(node.val, " ")
	preOrderTraverse1(node.left)
	preOrderTraverse1(node.right)
}

func postOrderTraverse1(node *Tree1) {
	if node == nil {
		return
	}
	postOrderTraverse1(node.left)
	postOrderTraverse1(node.right)
	fmt.Print(node.val, " ")
}

func main() {
	tree1 := &Tree1{}
	tree1 = insert1(tree1, 1)
	tree1 = insert1(tree1, 2)
	tree1 = insert1(tree1, 3)
	tree1 = insert1(tree1, 4)
	tree1 = insert1(tree1, 5)
	tree1 = insert1(tree1, 5)
	tree1 = insert1(tree1, 7)
	tree1 = insert1(tree1, 8)
	tree1 = insert1(tree1, 9)

	fmt.Print("Inorder traversal: ")
	inOrderTraverse1(tree1)
	fmt.Println()

	fmt.Print("PreOrder traversal: ")
	preOrderTraverse1(tree1)
	fmt.Println()

	fmt.Print("PostOrder traversal: ")
	postOrderTraverse1(tree1)
	fmt.Println()
}

File: trees/1.js
const root = Symbol('root');

class Node {
  constructor(data) {
    this.data = data;
    this.left = null;
    this.right = null;
  }
}

class bTree {
  constructor() {
    this[root] = null;
  }

  inOrderTraverse(node) {
    if (node == null) return;
    this.inOrderTraverse(node.left); 
    process.stdout.write(node.data + " ")
    this.inOrderTraverse(node.right); 
  }

  preOrderTraverse(node) {
    if (node == null) return;
    process.stdout.write(node.data + " ")
    this.preOrderTraverse(node.left);  
    this.preOrderTraverse(node.right); 
  }

  postOrderTraverse(node) {
    if (node == null) return;
    this.postOrderTraverse(node.left);  
    this.postOrderTraverse(node.right);
    process.stdout.write(node.data + " ")
  }

}

let btree = new bTree();

let tree = new Node(23, null, null);
tree.left = new Node(16, null, null);
tree.right = new Node(45, null, null);
tree.left.left = new Node(3, null, null);
tree.left.right = new Node(22, null, null);
tree.right.left = new Node(37, null, null);
tree.right.right = new Node(99, null, null);

process.stdout.write("Inorder traversal: "); 
btree.inOrderTraverse(tree); 
console.log();

process.stdout.write("PreOrder traversal: "); 
btree.preOrderTraverse(tree); 
console.log();

process.stdout.write("PostOrder traversal: "); 
btree.postOrderTraverse(tree); 
console.log();